apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -1108,15 +1098,37 @@ post_key_block(Config) ->
     post_key_block(?config(current_block_type, Config), Config).
 
 post_key_block(_CurrentBlockType, Config) ->
-    {ok, 200, PendingKeyBlock} = get_key_blocks_pending_sut(),
-    PowEvidence = lists:duplicate(42, 1),
-    Nonce = 1,
-    KeyBlock = PendingKeyBlock#{<<"pow">> => PowEvidence, <<"nonce">> => Nonce},
+    {ok, 200, #{<<"height">> := Height} = PendingKeyBlock} = get_key_blocks_pending_sut(),
+
+    KeyBlock = PendingKeyBlock#{<<"pow">> => lists:duplicate(42, 1), <<"nonce">> => 1},
     {ok, 400, Error} = post_key_blocks_sut(KeyBlock),
     %% Block is always rejected - pow and nonce are not correct.
     ?assertEqual(<<"Block rejected">>, maps:get(<<"reason">>, Error)),
+
+    KeyBlock1 = PendingKeyBlock#{<<"pow">> => lists:duplicate(42, 0), <<"nonce">> => 0},
+    {ok, KeyBlockHeader} = aec_headers:deserialize_from_client(key, KeyBlock1),
+    KeyBlockHeaderBin = aec_headers:serialize_to_binary(KeyBlockHeader),
+    Target = aec_headers:target(KeyBlockHeader),
+    Nonce = aec_pow:pick_nonce(),
+    {ok, {Nonce1, PowEvidence}} = mine_key_block(KeyBlockHeaderBin, Target, Nonce, 1000),
+    {ok, 200, #{}} = post_key_blocks_sut(PendingKeyBlock#{<<"pow">> => PowEvidence, <<"nonce">> => Nonce1}),
+    ok = aecore_suite_utils:wait_for_height(?config(node, Config), Height),
+    {ok, 200, CurrentBlock} = get_key_blocks_current_sut(),
+    ?assertEqual(Height, maps:get(<<"height">>, CurrentBlock)),
+    ?assertEqual(PowEvidence, maps:get(<<"pow">>, CurrentBlock)),
+    ?assertEqual(Nonce1, maps:get(<<"nonce">>, CurrentBlock)),
     ok.
 
+mine_key_block(HeaderBin, Target, Nonce, Attempts) when Attempts > 0 ->
+    case rpc(aec_mining, mine, [HeaderBin, Target, Nonce]) of
```

Comment:
I consider fine to mine the test block from the test on the system under test (I expect just the erlexec process(es) and the RPC worker for this request).